File: Frontend/HttpClient/src/app/models/order.ts
import { OrderQuantum } from './order-quantum';

export class Order {
    public id?: number | {};
    public name: string;
    public description: string;
    public state: OrderState;
    public plannedBeginDate: Date;
    public plannedEndDate: Date;
    public orderQuantums: OrderQuantum[];
    public orderQuantumsCount: number;

    static fromJSON(obj: any) {
        if (!obj) {
            return null;
        }

        return Object.assign(
            new Order(),
            obj
        );
    }
}
export enum OrderState {
    Undefined = 0,
    Ready = 1,
    InProcess = 2,
    Error = 3
}

File: Backend/Scheduler/Model/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Scheduler.Model.OrderReport;

namespace Scheduler.Model
{
    public enum OrderState
    {
        Undefined = 0,

        Ready = 1,

        InProcess = 2,

        Error = 3
    }

    public class Order
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public OrderState State { get; set; }

        public DateTime? PlannedBeginDate { get; set; }

        public DateTime? PlannedEndDate { get; set; }

        public virtual List<OrderQuantum> OrderQuantums { get; set; }

        public virtual List<OrderReport.OrderReport> OrderReports { get; set; }
    }
}
